rules/misc_rules.build_defs

```diff
@@ -451,41 +454,22 @@ def remote_file(name:str, url:str|list, hashes:list=None, out:str=None, binary:b
     urls = [url] if isinstance(url, str) else url
     url = urls[0]
 
-    urls_str = ' '.join(url) if isinstance(url, list) else url
-
     if bool(username) != bool(password_file):
         fail(f"Both or neither of username and password_file should be specified")
 
-    auth = username and password_file
-    curl = auth or headers or secret_headers
-    header_flag = ""
-    cmd = ''
-
-    if curl:
-        for header, value in headers.items():
-            header_flag = f'{header_flag} --header "{header}: {value}"'
-        for header, value in secret_headers.items():
-            header_flag = f'{header_flag} --header "{header}: $(cat {value})"'
-
-        # Authentication info
-        user_info = f'--user {username}:$(cat {password_file})' if auth else ""
+    labels = labels + [f"remote_file:header:{h}:{v}" for h, v in headers.items()]
```

Comment:
I didn't want to add more args to `build_rule()` so I'm doing this via labels

Reply:
The usage of `labels` is quite neat and extensible.